src/amount.cpp

```diff
@@ -9,16 +9,18 @@
 
 const std::string CURRENCY_UNIT = "BTC";
 
-CFeeRate::CFeeRate(const CAmount& nFeePaid, size_t nSize)
+CFeeRate::CFeeRate(const CAmount& nFeePaid, size_t nSize_)
 {
+    int64_t nSize = nSize_;
     if (nSize > 0)
-        nSatoshisPerK = nFeePaid*1000/nSize;
+        nSatoshisPerK = nFeePaid * 1000 / nSize;
```

Comment:
This would require a cast in 4 places instead of two (see diff below) but I am happy to apply the diff an squash. Please let me know how to proceed so we can finalize this.

```diff
diff --git a/src/amount.cpp b/src/amount.cpp
index 7b8618d..3966c5f 100644
--- a/src/amount.cpp
+++ b/src/amount.cpp
@@ -13,8 +13,7 @@ CFeeRate::CFeeRate(const CAmount& nFeePaid, size_t nBytes_)
 {
     assert(nBytes_ <= uint64_t(std::numeric_limits<int64_t>::max()));
-    int64_t nSize = int64_t(nBytes_);
 
-    if (nSize > 0)
-        nSatoshisPerK = nFeePaid * 1000 / nSize;
+    if (int64_t(nBytes_) > 0)
+        nSatoshisPerK = nFeePaid * 1000 / int64_t(nBytes_);
     else
         nSatoshisPerK = 0;
@@ -24,9 +23,8 @@ CAmount CFeeRate::GetFee(size_t nBytes_) const
 {
     assert(nBytes_ <= uint64_t(std::numeric_limits<int64_t>::max()));
-    int64_t nSize = int64_t(nBytes_);
 
-    CAmount nFee = nSatoshisPerK * nSize / 1000;
+    CAmount nFee = nSatoshisPerK * int64_t(nBytes_) / 1000;
 
-    if (nFee == 0 && nSize != 0) {
+    if (nFee == 0 && int64_t(nBytes_) != 0) {
         if (nSatoshisPerK > 0)
             nFee = CAmount(1);